File: scratch/inspect_formulas.js
const XLSX = require('xlsx');

const wb = XLSX.readFile('../master rekap 2026_Bom.xlsx');
const sheet = wb.Sheets['Data Validation'];
const data = XLSX.utils.sheet_to_json(sheet, { header: 1 });

const cols = ['W', 'X', 'Y', 'Z', 'AA', 'AB'];
const formulas = new Set();

for (let r = 7; r < data.length; r++) {
  const rowNum = r + 1;
  const codeVal = sheet['B' + rowNum]?.v;
  if (!codeVal) continue;
  const codeStr = codeVal.toString().trim();
  if (codeStr !== 'Parts' && codeStr !== 'Prt') continue;

  const valCell = sheet['D' + rowNum];
  if (!valCell || typeof valCell.v !== 'number' || valCell.v < 100) continue;

  const nameCell = sheet['C' + rowNum]?.v;

  let rowInfo = `Row ${rowNum}: Code=${valCell.v} Name="${nameCell}"`;
  let hasFormulas = false;
  cols.forEach(col => {
    const cell = sheet[col + rowNum];
    if (cell && cell.f) {
      rowInfo += ` | ${col}: "${cell.f}"`;
      formulas.add(`${col}: ${cell.f}`);
      hasFormulas = true;
    } else if (cell) {
      rowInfo += ` | ${col}: ${cell.v}`;
    }
  });
  if (hasFormulas) {
    console.log(rowInfo);
  }
}

console.log('\n--- Unique Formulas Found ---');
Array.from(formulas).sort().forEach(f => console.log(f));

File: scratch/inspect_formulas.py
import openpyxl

wb_path = '/Applications/Arenga/vscode/breakdown_app/master rekap 2026_Bom.xlsx'
wb = openpyxl.load_workbook(wb_path, data_only=False)
print("Sheets in 2026_Bom:", wb.sheetnames)

sheet = wb['Breakdown']
print("Successfully loaded sheet Breakdown")

# Let's inspect formulas in a few columns for rows 14-30
# Row 14, 15, 16, 17, etc.
cols = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'J', 'L', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', 'AA', 'AB', 'AC', 'AD', 'AE', 'AF', 'AG', 'AH', 'AI', 'AJ', 'AK', 'AL', 'AM', 'AN', 'AO', 'AP', 'AQ', 'AR', 'AS', 'AT', 'AU', 'AV', 'AW', 'AX', 'AY', 'AZ', 'BA', 'BB', 'BC', 'BD', 'BE', 'BF', 'BG', 'BH', 'BI', 'BJ', 'BK', 'BL', 'BM', 'BN', 'BO', 'BP', 'BQ', 'BR', 'BS', 'BT', 'BU', 'BV', 'BW', 'BX', 'BY', 'BZ', 'CA', 'CB', 'CC', 'CD', 'CE', 'CF']

print("Row 14 formulas:")
for col in cols:
    val = sheet[f'{col}14'].value
    if val and str(val).startswith('='):
        print(f"  Col {col}: {val}")

print("\nRow 15 formulas:")
for col in cols:
    val = sheet[f'{col}15'].value
    if val and str(val).startswith('='):
        print(f"  Col {col}: {val}")

print("\nRow 16 formulas:")
for col in cols:
    val = sheet[f'{col}16'].value
    if val and str(val).startswith('='):
        print(f"  Col {col}: {val}")
